parse-rest/journey-leg.js

```diff
@@ -57,7 +57,7 @@ const parseJourneyLeg = (ctx, l) => { // l = leg
 		res.tripId = l.JourneyDetailRef && l.JourneyDetailRef.ref || null
 
 		// todo: is it num instead of number?
-		const product = {name: l.name, number: l.number, ...l.Product}
+		const product = {name: l.name, number: l.number, ...(l.Product[0] || {})}
```

Comment:
I think this should be done in the [simplification/`findInTree` section](https://github.com/public-transport/hafas-client/blob/e6755ffbd18a0f1d1610cca1c0e6173b6fca7d98/rest-exe.js#L110-L128).